player.py

```diff
@@ -19,54 +26,56 @@ def sanitize(ytid):
 def path_for(ytid):
 	return os.path.join(DATA_DIR, sanitize(ytid) + ".mp4")
 
-def get_env():
-	env = dict(os.environ)
-	if display_video:
-		env["DISPLAY"] = ":0.0"
-	return env
-
 if display_video:
-	subprocess.check_call(os.path.join(os.path.dirname(os.path.abspath(__file__)), "configure-screen.sh"), env=get_env())
+	subprocess.check_call(os.path.join(os.path.dirname(os.path.abspath(__file__)), "configure-screen.sh"))
 
 def start_playing(uuid, ytid):
-	global current_uuid, current_subprocess
+	global current_uuid, should_be_paused
 	if current_uuid is not None:
 		stop_playing()
-	assert current_subprocess is None
+	assert player.filename is None
 	if os.path.exists(path_for(ytid)):
 		current_uuid = uuid
-		if display_video:
-			current_subprocess = subprocess.Popen(["mplayer", path_for(ytid), "-fs", "--xineramascreen=1"], env=get_env())
-		else:
-			current_subprocess = subprocess.Popen(["mplayer", path_for(ytid), "-vo", "null"], env=get_env())
+		player.loadfile(path_for(ytid))
+		should_be_paused = False
 
 def stop_playing():
-	global current_uuid, current_subprocess
-	assert current_uuid is not None and current_subprocess is not None
+	global current_uuid
+	assert current_uuid is not None
 	current_uuid = None
-	if current_subprocess.poll() is None:
-		current_subprocess.terminate()
-		if current_subprocess.poll() is None:
-			time.sleep(0.2)
-			if current_subprocess.poll() is None:
-				current_subprocess.kill()
-				current_subprocess.wait()
-	assert current_subprocess.poll() is not None
-	current_subprocess = None
-
-def check_on_process():
-	global current_uuid, current_subprocess
-	if current_subprocess is not None and current_subprocess.poll() is not None:
-		current_subprocess = None
+	player.stop()
+
+def playback_pause():
+	global should_be_paused
+	should_be_paused = not should_be_paused
+	player.pause()
+
+def check_finished_uuid():
+	global current_uuid
+	if player.filename is None:
 		uuid = current_uuid
 		current_uuid = None
 		return uuid
 	else:
 		return False
 
+def control_callback(message):
+	if player.filename is not None:
+		playback_pause()
+
+p = redis.pubsub(ignore_subscribe_messages=True)
+p.subscribe(**{'musicacontrol': control_callback})
```

Comment:
Why not `p.subscribe(musicacontrol=control_callback)`?

Reply:
Because cargo-cult following the python redis docs. I'll fix that.